src/pages/wallets/find-wallet.tsx

```diff
@@ -107,15 +108,21 @@ export const getStaticProps = (async ({ locale }) => {
 }) satisfies GetStaticProps<BasePageProps>
 
 const FindWalletPage = () => {
-  const randomizedWalletData = shuffle(walletData)
-  const { pathname } = useRouter()
+  const { pathname, locale } = useRouter()
   const theme = useTheme()
   const { t } = useTranslation("page-wallets-find-wallet")
   const resetWalletFilter = useRef(() => {})
   const { isOpen: showMobileSidebar, onOpen, onClose } = useDisclosure()
   const [filters, setFilters] = useState(filterDefault)
   const [selectedPersona, setSelectedPersona] = useState(NaN)
 
+  // If any wallet supports user's locale, show them (shuffled) at the top and then the remaining ones
+  const supportedLocaleWallets = getSupportedLocaleWallets(locale!)
+  const noSupportedLocaleWallets = getNonSupportedLocaleWallets(locale!)
+  const randomizedWalletData = supportedLocaleWallets.concat(
+    noSupportedLocaleWallets
+  )
```

Comment:
Since we are doing a `shuffle`, we will get a different result on each render, causing a hydration warning. We should only do this on client side (maybe inside a `useEffect`).